### PR TITLE
Fix npm dependencies and shrinkwrap for production deploys

### PR DESCRIPTION
**Pull request by willbarton**
Our gulp tasks currently require some dependencies that are listed in package.json as `devDependencies`. This is a problem when we run a production install, or try to shrinkwrap a production install. This PR adds those dependencies to `dependencies` in package.json and creates a new production shrinkwrap. It also ensures that optional dependencies won't be installed in production. 

This is needed to successfully deploy in production.

## Changes

- Included build dependencies in package.json

## Testing

```
rm -rf node_modules
./frontend.sh production
```

This should succeed.

## Notes

- We should refactor the build gulp tasks to require only the dependencies they need.

## Checklist

* [ ] Changes are limited to a single goal (no scope creep)
* [ ] Code can be automatically merged (no conflicts)
* [ ] Code follows the standards laid out in the [front end playbook](https://github.com/cfpb/front-end)
* [ ] Passes all existing automated tests
* [ ] New functions include new tests
* [ ] New functions are documented (with a description, list of inputs, and expected output)
* [ ] Placeholder code is flagged
* [ ] Visually tested in supported browsers and devices
* [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
* [ ] Reviewers requested with the [Assignee tool](https://help.github.com/articles/assigning-issues-and-pull-requests-to-other-github-users/) :arrow_right:

### CODE REVIEW

**Comment by sebworks** [comment]
@willbarton, are the `devDependencies` already installed on Jenkins?

File: package.json [hunk 1]
```diff
@@ -33,27 +33,28 @@
...
+    "gulp-mocha": "3.0.1",
     "is-reachable": "1.3.0",
+    "gulp-util": "3.0.8"
+  },
+  "devDependencies": {
+    "chai": "3.5.0",
+    "gulp-plumber": "1.1.0",
     "jasmine-reporters": "2.1.1",
     "jasmine-spec-reporter": "2.4.0",
     "jquery": "1.11.3",
```
Comment (sebworks) [inline package.json]: `jquery` should get moved to dependencies. We still require it for a CDN fallback in `base.html`.

File: package.json [hunk 2]
```diff
@@ -33,27 +33,28 @@
...
-    "gulp-util": "3.0.7",
+    "gulp-coveralls": "0.1.4",
+    "gulp-istanbul": "1.1.1",
+    "gulp-mocha": "3.0.1",
     "is-reachable": "1.3.0",
+    "gulp-util": "3.0.8"
+  },
+  "devDependencies": {
+    "chai": "3.5.0",
+    "gulp-plumber": "1.1.0",
```
Comment (sebworks) [inline package.json]: Doesn't look like we still use `gulp-plumber`.

**Comment by willbarton** [comment]
I will also note, I purposefully left a space between the old dependencies and the new ones so that we can easily differentiate when refactoring.

**Comment by willbarton** [comment]
@sebworks the devDependencies are not already installed on Jenkins.

**Comment by willbarton** [comment]
Confirmed the latest change work in production Jenkins as well.

**Comment by sebworks** [comment]
> @sebworks the devDependencies are not already installed on Jenkins in production.

ok, I just wasn't sure if we wanted to run any tests on Jenkins.  The way it's set up now all tests run on Travis.

**Comment by willbarton** [comment]
@sebworks Yeah. We should set it up to run tests on Jenkins, but we can create a job that works the same way as Travis. It would be separate from the build job though.

**Pull request merged**